File: tests/snapshots/format.py
# Format uses a Rust format string, not VB's #.### pictures.

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    print(_vb('{:.2f}'.format(3.14159)))
    print(_vb('{:04}'.format(7)))
    print(_vb('approx {:.1f}'.format(3.5)))


if __name__ == "__main__":
    main()

File: tests/snapshots/format.rs
// Format uses a Rust format string, not VB's #.### pictures.

fn vbr_main() -> Result<(), String> {
    println!("{}", format!("{:.2}", 3.14159));
    println!("{}", format!("{:04}", 7));
    println!("{}", format!("approx {:.1}", 3.5));
    Ok(())
}

fn main() {
    if let Err(error) = vbr_main() {
        eprintln!("Error: {error}");
        std::process::exit(1);
    }
}
